Dockerfile

```diff
@@ -0,0 +1,17 @@
+ARG UBUNTU_VERSION=22.04
+
+FROM ubuntu:$UBUNTU_VERSION
+
+RUN apt-get update && \
+    apt-get install -y build-essential python3 python3-pip
+
+RUN pip install --upgrade pip setuptools wheel \
+    && pip install torch torchvision torchaudio sentencepiece numpy
```

Comment:
I believe the Python install is only needed to convert the model files, maybe that could be moved to a different Dockerfile so this one can stay smaller? (I can also see it making sense to keep things simple and only have one Dockerfile for everything though)

Reply:
Thanks for your report.

Splitted into two stages (build & runtime)

Reply:
As far as I can tell the Makefile doesn’t even use Python: https://github.com/ggerganov/llama.cpp/blob/master/Makefile

Reply:
Sorry, I didn't understand you the first time.

I have separated into two Dockerfiles, one for the "tools" and another for the main.

To test:
1) build new images:
tools:
`docker build -f .devops/tools.Dockerfile -t llamacpp-converter . `
main:
`docker build -f .devops/main.Dockerfile -t llamacpp-main . `

2) usage:
convert model 7B pth into ggml
`docker run -v models:/models llamacpp-converter "/models/7B/" 1`
execute main process:
`docker run -v models:/models llamacpp-main -m /models/7B/ggml-model-q4_0.bin -p "Building a website can be done in 10 simple steps:" -t 8 -n 512`